src/Symfony/Component/Utf8/Bytes.php

```diff
@@ -0,0 +1,505 @@
+
+        return $bytes;
+    }
+
+    public static function fromCharCode(int ...$codes): self
+    {
+        $bytes = new self();
+        $bytes->string = implode('', array_map('chr', $codes));
+
+        return $bytes;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function __toString()
+    {
+        return $this->string;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function count(): int
+    {
+        return strlen($this->string);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function length(): int
+    {
+        return strlen($this->string);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isEmpty(): bool
+    {
+        return '' === $this->string;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function explode(string $delimiter, int $limit = null): array
+    {
+        if ('' === $delimiter) {
+            throw new InvalidArgumentException('Passing an empty delimiter is not supported by this method. Use getIterator() method instead.');
+        }
+
+        $chunks = array();
+        foreach (explode($delimiter, $this->string, $limit ?: PHP_INT_MAX) as $i => $string) {
+            $chunks[$i] = $chunk = clone $this;
+            $chunk->string = $string;
+        }
+
+        return $chunks;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function toLowerCase(): self
+    {
+        $current = setlocale(LC_CTYPE, '0');
+        setlocale(LC_CTYPE, 'C');
+
+        $result = clone $this;
+        $result->string = strtolower($this->string);
+
+        setlocale(LC_CTYPE, $current);
+
+        return $result;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function toUpperCase(): self
+    {
+        $current = setlocale(LC_CTYPE, '0');
+        setlocale(LC_CTYPE, 'C');
+
+        $result = clone $this;
+        $result->string = strtoupper($this->string);
+
+        setlocale(LC_CTYPE, $current);
+
+        return $result;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function toUpperCaseFirst($allWords = false)
+    {
+        $current = setlocale(LC_CTYPE, '0');
+        setlocale(LC_CTYPE, 'C');
+
+        $result = clone $this;
+        $result->string = $allWords ? ucwords($this->string) : ucfirst($this->string);
+
+        setlocale(LC_CTYPE, $current);
+
+        return $result;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function toFoldedCase(bool $full = true)
+    {
+        return $this->toLowerCase();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function substr(int $start = 0, int $length = null): self
+    {
+        $result = clone $this;
+        $result->string = substr($this->string, $start, $length);
+
+        return $result;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function trim(string $charsList = null): self
+    {
+        $result = clone $this;
+        $result->string = trim($this->string, $charsList ?: " \t\n\r\0\x0B");
```

Comment:
Neither can `Bytes` because we cannot know which bytes are whitespace, since we cannot know what kind of character set we are dealing with. That is another point that I am making in multiple comments. 😉